components/brave_wallet/browser/solana_provider_impl.cc

```diff
@@ -112,11 +112,11 @@ void SolanaProviderImpl::Connect(absl::optional<base::Value::Dict> arg,
     return;
   }
 
-  delegate_->IsAccountAllowed(
-      mojom::CoinType::SOL, *account,
-      base::BindOnce(&SolanaProviderImpl::ContinueConnect,
-                     weak_factory_.GetWeakPtr(), is_eagerly_connect, *account,
-                     std::move(callback)));
+  const bool allowed =
+      delegate_->IsAccountAllowed(mojom::CoinType::SOL, *account);
+
+  ContinueConnect(is_eagerly_connect, *account, std::move(callback), allowed);
```

Comment:
Since now we don't have async IsAccountAllowed, we can eliminate `ContinueConnect`

Reply:
https://github.com/brave/brave-core/pull/18358/commits/7b1a3f40969e221ce851b61f06499322bc2fc6f6